iree/tools/opt_main.cc

```diff
@@ -68,6 +69,7 @@ int main(int argc, char **argv) {
   mlir::iree_compiler::registerIreeCompilerModuleDialects();
   mlir::iree_compiler::registerAllIreePasses();
   mlir::iree_compiler::registerHALTargetBackends();
+  mlir::iree_compiler::registerSPRIVTranslation();
```

Comment:
Wonder if we'd want a `registerAllIreeTranslations()` in `iree/tools/` when there are more translations in https://github.com/google/iree/tree/master/iree/compiler/Translation . It could call this `registerSPRIVTranslation`.

Reply:
> Are any changes needed to the python bindings? We don't have presubmit coverage for those on GitHub (yet).

I don't think so. I build and tested locally with CMake.

Reply:
Well, I did not yet go through all the translations and I am not yet aware if we really need a `registerAllIreeTranslations()` function or if this would pull in deps not necessary. Most targets are required by `iree-opt`
https://github.com/google/iree/blob/c931490e9c7ca4aa4057be669a85993ea4778b7e/iree/tools/CMakeLists.txt#L216-L219
but `iree-translate`
https://github.com/google/iree/blob/c931490e9c7ca4aa4057be669a85993ea4778b7e/iree/tools/CMakeLists.txt#L290-L291
for example does not depend on `CodegenPasses`. Fruther `iree-run-mlir` only depends on `iree::compiler::Translation::IREEVM`. If there is a need for a `registerAllIreeTranslations()` function, we can still add this later, but I would like to avoid to implement an unused function.

Reply:
Yeah, let's keep it simple for now.

Would it be useful to eventually document how registration is set up with some developer-facing docs like https://github.com/google/iree/blob/master/iree/compiler/Dialect/VMLA/README.md#adding-an-op ? "Adding a new [pass, backend, translation, etc.]"

Reply:
Probably it would be useful!